frontend/src/metabase/containers/ItemPicker/ItemPicker.tsx

```diff
@@ -243,8 +256,10 @@ function ItemPicker<TId>({
         checkHasWritePermissionForItem={checkHasWritePermissionForItem}
         getCollectionIcon={getCollectionIcon}
         style={style}
-        // personal is a fake collection for admins that contains all other user's collections
-        allowFetch={openCollectionId !== "personal"}
+        allowFetch={
+          // "personal" is a fake collection for admins that contains all other user's collections
+          openCollectionId !== "personal" && !hideItems
```

Comment:
This way, we don't need to modify `GET /api/collection/:id/items` to exclude items from public collections because we know from the start we don't want to render any items.

Reply:
And one more thing. It's usually easier to follow boolean variable names if they are in "positive" form. E.g. `showItems`, `allowFetch`, `canShowItems` etc. and not `hideItems`, `disableFetch`

Reply:
But `hideItems` consists of 3 other conditions
```ts
!searchString &&
showOnlyPersonalCollections &&
!isOpenCollectionInPersonalCollection
```

It might not be apparent why `searchString` is needed, because we're using `Search.ListLoader` for 2 scenarios:
1. when we're not searching:
  - we want `allowFetch` to be false when we're viewing public collections (only root collection in this case) and is on a question in a personal collection.
2. when we are searching:
  - we want `allowFetch` to be `true`, so `Search.ListLoader` could search the dashboards.

So I'm worried inlining those 3 conditions might make it harder to understand. What about renaming `hideItems` to `showItems` and flip the condition instead?

Reply:
Fixed in 2f662c4d5aa75f2521aa422ff750cc21aa12ef3f